File: src/js/src/strategies/WriteEscapedBracketStrategy.js
function WriteEscapedBracketStrategy() {
	WriteEscapedBracketStrategy.$parent.call(this);
	
	this.addSatisfier(new InScopeStackSatisfier(ScopeTypeCache.link, ScopeTypeCache.module, ScopeTypeCache.Unescaped, ScopeTypeCache.raw));
	this.setAsTokenTransformer("]]");
	
	this.getChar = function(context) { return "]"; };
	
	this.priority = 10000;
}

extend(WriteCharacterStrategy, WriteEscapedBracketStrategy);

File: src/dotnet/Src/ButterflyNet.Parser/Strategies/WriteEscapedBracketStrategy.cs
using ButterflyNet.Parser.Satisfiers;

namespace ButterflyNet.Parser.Strategies {
	[TokenTransformer("]]")]
	public class WriteEscapedBracketStrategy : WriteCharacterStrategy {
		public WriteEscapedBracketStrategy() {
			AddSatisfier(new InScopeStackSatisfier(ScopeTypeCache.Link, ScopeTypeCache.Module, ScopeTypeCache.Unescaped, ScopeTypeCache.Raw));
		}

		protected override char GetChar(ParseContext context) {
			return ']';
		}

		public override int Priority { get { return base.Priority - 1; } }
	}
}